tests/test_license_plate.py

```diff
@@ -1,26 +1,141 @@
+                mock__is_valid_mercosul.assert_not_called()
+
+        # Mock _is_valid_old_format return False
+        with patch(
+            "brutils.license_plate._is_valid_mercosul"
+        ) as mock__is_valid_mercosul:
+            with patch(
+                "brutils.license_plate._is_valid_old_format", return_value=False
+            ) as mock__is_valid_old_format:
+                # When licence plate old format isn't valid, returns False
+                self.assertIs(is_valid("123456", "old_format"), False)
+
+                # Checks if function _is_valid_old_format is called with the
+                # correct argument
+                mock__is_valid_old_format.assert_called_once_with("123456")
+
+                # Checks if function _is_valid_mercosul is not called
+                mock__is_valid_mercosul.assert_not_called()
+
+    def test_is_valid_when_type_is_mercosul(self):
+        # Mock _is_valid_mercosul to return True
+        with patch(
+            "brutils.license_plate._is_valid_old_format",
+        ) as mock__is_valid_old_format:
+            with patch(
+                "brutils.license_plate._is_valid_mercosul", return_value=True
+            ) as mock__is_valid_mercosul:
+                # When license plate as mercosul is_valid, returns True
+                self.assertIs(is_valid("ABC4E67", "mercosul"), True)
+
+                # Checks if function _is_valid_mercosu is called with the
+                # correct argument
+                mock__is_valid_mercosul.assert_called_once_with("ABC4E67")
+
+                # Checks if function _is_valid_old_format is not called
+                mock__is_valid_old_format.assert_not_called()
+
+        # Mock _is_valid_mercosul to return False
+        with patch(
+            "brutils.license_plate._is_valid_old_format",
+        ) as mock__is_valid_old_format:
+            with patch(
+                "brutils.license_plate._is_valid_mercosul", return_value=False
+            ) as mock__is_valid_mercosul:
+                # When mercosul plate is not valid, returns False
+                self.assertIs(is_valid("11994029275", "mercosul"), False)
+
+                # Checks if function _is_valid_mercosul is called with the
+                # correct argument
+                mock__is_valid_mercosul.assert_called_once_with("11994029275")
+
+                # Checks if function _is_valid_old_format is not called
+                mock__is_valid_old_format.assert_not_called()
+
+    def test_is_valid_when_type_is_none(self):
+        # Mock _is_valid_mercosul to return True
+        with patch(
+            "brutils.license_plate._is_valid_mercosul", return_value=True
+        ) as mock__is_valid_mercosul:
+            with patch(
+                "brutils.license_plate._is_valid_old_format"
+            ) as mock__is_valid_old_format:
+                # When license plate as mercosul is_valid, returns True
+                self.assertIs(is_valid("ABC4E67"), True)
```

Comment:
Como a função `is_valid` retorna `_is_valid_old_format(…) or _is_valid_mercosul(…)`, se não retornar algo q se pareça com `False`, nunca vamos chegar no `_is_valid_mercosul(…)`.

Como vc não definiu o retorno do _mock_ de `_is_valid_old_format`, ele retorna um objeot `Mock()` que não se parece com `False`. Por isso nunca executamos a função do Mercosul como esperado : )

```suggestion
            with patch(
                "brutils.license_plate._is_valid_old_format", return_value=False
            ) as mock__is_valid_old_format:
                # When license plate as mercosul is_valid, returns True
                self.assertIs(is_valid("ABC4E67"), True)
```